### DISCUSSION TITLE
No plugin deployment on mingw64 and Compilation Error for calloc parameters inversion.

### DISCUSSION THREAD

**Discussion by Cedric-29200** [QuasarApp/CQtDeployer#831] [Q&A]
Hello, 
because windeployqt doesn't work anymore with mingw, I wanted to try CQtDeploy. I managed to make it work on my configuration (mingw with gcc compiler up to date 14.2) but I encountered two issues :

Most annoying one is that when using CQtDeploy I have the message :
> "Failed to execute the qmake process!"

I found out that snapRootFS() is empty for mingw (windows) so maybe the program could check this and still call setQmake(qmake).

Other small problem:
Everywhere in the code we have calloc used in this expected form : `calloc(size_t n, sizeof(struct X))` except in zip.c where compilation fails with recent gcc compiler.

> zip.c:1676:24: error: 'calloc' sizes specified with 'sizeof' in the earlier argument and not in the later argument [-Werror=calloc-transposed-args]
>  1676 |   *buf = calloc(sizeof(unsigned char), n);

Not a big deal as multiplication is commutative but just wanted to mention it.

Thank you for this wonderful program.

**Comment by EndrII** [comment]
@Cedric-29200 thanks you for your feedback ) 
about snapRootFS this is snap fs prefix, so for windows or not snap-packed version of cqtdeployer it is empty path.  
But cqtdeployer should not invoke this function for not snap packages, it looks as issue. Please attach your full deploy log. 

About zip. We use external lib to package zip arrives, I can add support dynamic excluding of this library from project for old GCC compilers if you want. 

i also will be happy if you help me fix your issues, if sure how we can fix it, you can create Pull Request with your fixes.
